programs/vote/benches/vote_instructions.rs

```diff
@@ -681,6 +682,320 @@ impl BenchUpdateVoteState {
+            serialize(&VoteInstruction::CompactUpdateVoteStateSwitch(
+                vote_state_update,
+                Hash::default(),
+            ))
+            .unwrap()
+        } else {
+            serialize(&VoteInstruction::CompactUpdateVoteState(vote_state_update)).unwrap()
+        };
+
+        let transaction_accounts = vec![
+            (vote_pubkey, vote_account.clone()),
+            (sysvar::slot_hashes::id(), slot_hashes_account.clone()),
+            (sysvar::clock::id(), create_default_clock_account()),
+        ];
+
+        Self {
+            instruction_data,
+            transaction_accounts,
+            instruction_accounts,
+        }
+    }
+    fn run(&self) {
+        let _accounts = process_deprecated_instruction(
+            &self.instruction_data,
+            self.transaction_accounts.clone(),
+            self.instruction_accounts.clone(),
+            Ok(()),
+        );
+    }
+}
+
+struct BenchTowerSync {
+    instruction_data: Vec<u8>,
+    transaction_accounts: Vec<(Pubkey, AccountSharedData)>,
+    instruction_accounts: Vec<AccountMeta>,
+}
+
+impl BenchTowerSync {
+    fn new(switch: bool) -> Self {
+        let (vote_pubkey, vote_account) = create_test_account();
+        let vote = Vote::new(vec![1], Hash::default());
+        let slot_hashes = SlotHashes::new(&[(*vote.slots.last().unwrap(), vote.hash)]);
+        let slot_hashes_account = account::create_account_shared_data_for_test(&slot_hashes);
+        let instruction_accounts = vec![
+            AccountMeta {
+                pubkey: vote_pubkey,
+                is_signer: true,
+                is_writable: true,
+            },
+            AccountMeta {
+                pubkey: sysvar::slot_hashes::id(),
+                is_signer: false,
+                is_writable: false,
+            },
+            AccountMeta {
+                pubkey: sysvar::clock::id(),
+                is_signer: false,
+                is_writable: false,
+            },
+        ];
+        let tower_sync = TowerSync::from(vec![(1, 1)]);
+
+        let instruction_data = if switch {
+            serialize(&VoteInstruction::TowerSyncSwitch(
+                tower_sync,
+                Hash::default(),
+            ))
+            .unwrap()
+        } else {
+            serialize(&VoteInstruction::TowerSync(tower_sync)).unwrap()
+        };
+
+        let transaction_accounts = vec![
+            (vote_pubkey, vote_account.clone()),
+            (sysvar::slot_hashes::id(), slot_hashes_account.clone()),
+            (sysvar::clock::id(), create_default_clock_account()),
+        ];
+
+        Self {
+            instruction_data,
+            transaction_accounts,
+            instruction_accounts,
+        }
+    }
+    fn run(&self) {
+        let _accounts = process_deprecated_instruction(
+            &self.instruction_data,
+            self.transaction_accounts.clone(),
+            self.instruction_accounts.clone(),
```

Comment:
nit: don't think `clone` takes much time here, just fyi, if really want to be accurate, can move accounts clone to `bencher.iter_with_setup` clause to exclude clone out of benching results